docs/pages/tv-settings_halSpec.md

```diff
@@ -0,0 +1,514 @@
+- `HAL` Tests will endeavour to create worst case scenarios to assist investigations.
+- Copyright validation is required to be performed, e.g.: `Black duck`, `FossID`.
+- Improvements by any party to the testing suite are required to be fed back.
+
+### Licensing
+
+This interface is expected to get released under the Apache License 2.0.
+
+### Build Requirements
+
+TV Settings `HAL` source code must build into a shared library and must be named as `libtvsettings-hal.so`.
+  
+### Variability Management
+
+Any changes in the `APIs` must be reviewed and approved by component architects.
+
+### Platform or Product Customization
+
+Product or platform specification requirements will be handled in vendor specific config file.
+
+## Interface API Documentation
+
+`API` documentation will be provided by Doxygen which will be generated from the header file(s).
+
+### Theory of operation and key concepts
+
+This interface handles various functionalities/requests related to Picture Quality settings :
+
+- Brightness
+- Contrast
+- Hue
+- Saturation
+- White Balance
+- Sharpness
+- Color Temperature
+- Backlight 
+- Aspect Ratio
+- Dimming Modes
+- Local Dimming Level
+- Low Latency state
+- Notify Video Format Change
+- Notify Video Resolution Change
+- Notify Video FrameRate Change
+- Notify Video Content Change
+
+There are other platform specific Picture Quality settings that can be managed by this interface :
+
+- CMS
+- Dolby Vision
+
+### Diagrams
+
+#### Operational Call Diagram
+##### Init and Callback Sequence
+```mermaid
+sequenceDiagram
+participant Caller as Caller
+    participant HAL as TV Settings HAL
+    participant OPPDB as Override Picture Profile DB
+    participant DPPDB as Default Picture Profile DB
+    participant Driver as SoC
+
+    Caller->>HAL:tvInit()
+    Note over HAL: Initialize the TV Setting HAL APIs
+    HAL->>Driver: Allocates resources
+    Driver-->>HAL:return
+    HAL->>DPPDB: Read default picture profile properites
+    DPPDB-->>HAL:return
+    HAL->>OPPDB: Copy default picture profile properites
+    OPPDB-->>HAL:return
+    HAL-->>Caller:return
+    Caller->>HAL: SetTVPictureMode()
+    Note over HAL: Set the default picture mode entertainment
+    HAL->>OPPDB: Read associated picture properties
+    OPPDB-->>HAL:return
+    Note over HAL: Reload gamma and white balance if there is a colour temperature value change.
+    HAL->>OPPDB: Read associated Gamma and White balance
+    OPPDB-->>HAL:return
+    Note over HAL: Reload TMAX if there is a LDIM level change
+    HAL->>OPPDB: Read associated TMAX value
+    OPPDB-->>HAL:return
+    Note over HAL: Apply the new picture properties, gamma, TMAX and white balance if they have changed.
+    HAL->>Driver: Apply new picture properties
+    Driver-->>HAL:return
+    HAL->>OPPDB: Update the Picture mode association table
+    Note over HAL: Associate new picture mode to current video format and current video source
+    OPPDB-->>HAL:return
+    HAL-->>Caller:return

```

Comment:
Just reconfirming gain. Does these changes Picture parameters will get erased during the init() function call again?  how will be recover if these picture values has been modified in the previous instance (before init) and when rebooted all those stored values are gone and replaced with default ones.

Reply:
its plugin responsibility to update with user changed values after TV init

Reply:
So plugin will maintain the same data base with respect to the picture parameters again similar to the kind of tables we have here?

Reply:
checked the architecture.. looks TR181 wll maintain these params.